pkg/resolver/related.go

```diff
@@ -342,6 +357,39 @@ func (s *SearchResult) updateKindMap(uid string, kind string, levelMap map[strin
 	levelMap[kind] = uids
 }
 
+// map the related result uids to the uids in the search input
+func (s *SearchResult) updResultToCurrSearchUidsMap(resultUid string, currSearchUidsMap map[string]struct{},
+	resultToCurrSearchUidsMap map[string][]string, path [2]string) {
+
+	for _, relatedUid := range path {
+		if _, ok := currSearchUidsMap[relatedUid]; ok {
+			klog.V(9).Infof("uid %s is valid and part of the current search. Number of relatedUids: %d",
+				relatedUid, len(resultToCurrSearchUidsMap[resultUid]))
+			if _, found := resultToCurrSearchUidsMap[resultUid]; !found {
+				resultToCurrSearchUidsMap[resultUid] = []string{relatedUid}
+			} else {
+				// if the relatedUid is not already added, append it
+				if !checkIfInArray(resultToCurrSearchUidsMap[resultUid], relatedUid) {
+					resultToCurrSearchUidsMap[resultUid] = append(resultToCurrSearchUidsMap[resultUid], relatedUid)
+					klog.V(9).Infof("uid %s is newly mapped to uids %+v", resultUid, resultToCurrSearchUidsMap[resultUid])
+				}
+			}
+			klog.V(9).Infof("uid %s is  related to uids %+v.", resultUid, resultToCurrSearchUidsMap[resultUid])
+			// Need to process only one uid in the path as it is either the sourceid or the destid that can be related
+			break
+		}
+	}
+}
+
+func checkIfInArray(lookupMap []string, uid string) bool {
```

Comment:
Just FYI, go 1.21 will add a new standard package `slices` to simplify this. It also has a `slices.Sort()` which we have somewhere else in this project.
https://pkg.go.dev/slices